### ISSUE TITLE
Update our action encoding to allow battlecry targets.

### ISSUE THREAD

**Issue by jeremysalwen** [JDBumgardner/stone_ground_hearth_battles#15]
Our current pytorch action encoding does not allow our bots to summon cards with battlecries.  We need a way to encode the battlecry target, and to calculate the probability of a card to summon + battlecry target and take its derivative.

I am thinking that we could first calculate a probability distribution using a softmax over all cards to play, then once we chose the card, we could feed that sampled choice back to the network to have it choose the target.

**Comment by jeremysalwen** [comment]
Also known as a pointer network (see the alphastar paper for an example of how its used).
